src/tests/resource-listers/AllResourcesLister.test.js

```diff
@@ -0,0 +1,210 @@
+            { TableNames: [ 'table-d', 'table-e', 'table-f' ], LastEvaluatedTableName: 'further-table-set' },
+            { TableNames: [ 'table-g', 'table-h', 'table-i' ] },
+         ]));
+
+         return lister._fetchTableNames()
+            .then(function(tableNames) {
+               sinon.assert.calledOnce(stub);
+               sinon.assert.calledOn(stub, lister);
+               sinon.assert.calledWith(stub, dynamo, 'listTables', {}, 'ExclusiveStartTableName', 'LastEvaluatedTableName');
+
+               expect(tableNames).to.eql([
+                  'table-a', 'table-b', 'table-c',
+                  'table-d', 'table-e', 'table-f',
+                  'table-g', 'table-h', 'table-i',
+               ]);
+
+               stub.restore();
+            });
+      });
+
+   });
+
+   describe('_fetchAll', function() {
+
+      it('works when there are only one page of results', function() {
+         var service = { action: function() { } },
+             actionStub = sinon.stub(service, 'action'),
+             expectedResponse = { data: 'value' };
+
+         actionStub.callsArgWithAsync(1, null, expectedResponse);
+
+         return lister._fetchAll(service, 'action', { search: 'param' }, 'requestToken', 'responseToken')
+            .then(function(resps) {
+               sinon.assert.calledOnce(actionStub);
+               sinon.assert.calledOn(actionStub, service);
+               sinon.assert.calledWith(actionStub, { search: 'param', requestToken: null });
+
+               expect(resps).to.be.an('array');
+               expect(resps).to.have.length(1);
+               expect(resps[0]).to.eql(expectedResponse);
+
+               actionStub.restore();
+            });
+      });
+
+      it('works when there is more than one page of results', function() {
+         var service = { action: function() { } },
+             actionStub = sinon.stub(service, 'action'),
+             expectedResponse1 = { data: 'some stuff', responseToken: 'get-more' },
+             expectedResponse2 = { data: 'more stuff' };
+
+         actionStub.onCall(0).callsArgWithAsync(1, null, expectedResponse1);
+         actionStub.onCall(1).callsArgWithAsync(1, null, expectedResponse2);
+
+         return lister._fetchAll(service, 'action', { search: 'param' }, 'requestToken', 'responseToken')
+            .then(function(resps) {
+               sinon.assert.calledTwice(actionStub);
+               sinon.assert.alwaysCalledOn(actionStub, service);
+
+               sinon.assert.calledWith(actionStub, { search: 'param', requestToken: null });
+               sinon.assert.calledWith(actionStub, { search: 'param', requestToken: 'get-more' });
+               sinon.assert.callOrder(
+                  actionStub.withArgs({ search: 'param', requestToken: null }),
+                  actionStub.withArgs({ search: 'param', requestToken: 'get-more' })
+               );
```

Comment:
Curiosity thing - I think I've done the same thing: lines 127 through 132 look very redundant, right? Like we're testing how `actionStub` was called, and then we're testing it again, just to test the order. Could the first tests just be deleted? Are the second tests (the `callOrder` tests) accomplishing the same thing? I suspect they are, but would need to look into sinon's assertions to verify. You want to take a look at that and de-dupe if we can somehow?